src/rest/impl/contracts.service.ts

```diff
@@ -109,35 +122,84 @@ export default class ContractsServiceImpl extends ContractsService {
     const rejectComment = contract.rejectComment;
     const proposedDeliverAll = contract.proposedDeliverAll;
 
-    const databaseContract = await models.createContract(userId, 
+    const databaseContract = await models.createContract(
+      userId,
       year,
-      deliveryPlaceId, 
-      proposedDeliveryPlaceId, 
-      itemGroupId, 
-      sapId, 
-      contractQuantity, 
-      deliveredQuantity, 
-      proposedQuantity, 
-      startDate, 
-      endDate, 
-      signDate, 
-      termDate, 
-      status, 
-      areaDetails ? JSON.stringify(areaDetails) : "",
+      deliveryPlaceId,
+      proposedDeliveryPlaceId,
+      itemGroupId,
+      sapId,
+      contractQuantity,
+      deliveredQuantity,
+      proposedQuantity,
+      startDate,
+      endDate,
+      signDate,
+      termDate,
+      status,
+      areaDetails ? JSON.stringify(areaDetails) : null,
       deliverAll,
       proposedDeliverAll,
-      remarks, 
-      deliveryPlaceComment, 
-      quantityComment, 
-      rejectComment);
+      remarks,
+      deliveryPlaceComment,
+      quantityComment,
+      rejectComment
+    );
 
-    if (databaseContract.status === "DRAFT") {
+    if (contract.status === "DRAFT") {
```

Comment:
Move DRAFT and APPROVED logic to own functions

Reply:
Draft logic is just one function call so it should be just here. Approved can be moved, though it can make things a bit unclear since it updates database and SAP, and the request is _possibly_ responded inside that too, so it can be hard to understand, what updates and when. I can try to refactor this.